engine/evarutil.mli

```diff
@@ -22,12 +22,22 @@ open EConstr
 val new_meta : unit -> metavariable
 val mk_new_meta : unit -> constr
 
+(** General evar naming using intro patterns  *)
+type intro_pattern_naming_expr =
+  | IntroIdentifier of Id.t
+  | IntroFresh of Id.t
+  | IntroAnonymous
+
```

Comment:
This should probably be put into Namegen, but I don't have a strong opinion on that.

Reply:
Sounds reasonable, let me see how things would look

Reply:
I looks better indeed, pushing the change to `Namegen`.